.github/workflows/backend.yml

```diff
@@ -0,0 +1,445 @@
+      - uses: actions/checkout@v2
+
+        with:
+          # Avoid codecov error message related to SHA resolution:
+          # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
+          fetch-depth: '2'
+
+      - name: Setup sentry env (python ${{ matrix.python-version }})
+        uses: ./.github/actions/setup-sentry
+        if: needs.changes.outputs.backend == 'true'
+        id: setup
+        with:
+          python-version: ${{ matrix.python-version }}
+          pip-cache-version: ${{ secrets.PIP_CACHE_VERSION }}
+          snuba: true
+
+      - name: Run backend test (${{ steps.setup.outputs.matrix-instance-number }} of ${{ steps.setup.outputs.matrix-instance-total }})
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          # Note: `USE_SNUBA` is not used for backend tests because there are a few failing tests with Snuba enabled.
+          unset USE_SNUBA
+          make test-python-ci
+
+      - name: Handle artifacts
+        uses: ./.github/actions/artifacts
+
+  cli:
+    needs: files-changed
+    name: cli test
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup sentry env (python ${{ matrix.python-version }})
+        uses: ./.github/actions/setup-sentry
+        id: setup
+        if: needs.changes.outputs.backend == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+          pip-cache-version: ${{ secrets.PIP_CACHE_VERSION }}
+
+      - name: Run test
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          make test-cli
+
+      - name: Handle artifacts
+        uses: ./.github/actions/artifacts
+
+  lint:
+    needs: files-changed
+    name: backend lint
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Internal github app token
+        id: token
+        uses: getsentry/action-github-app-token@v1
+        continue-on-error: true
+        with:
+          app_id: ${{ secrets.SENTRY_INTERNAL_APP_ID }}
+          private_key: ${{ secrets.SENTRY_INTERNAL_APP_PRIVATE_KEY }}
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: ./.github/actions/setup-python
+        if: needs.changes.outputs.backend == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+          # Note this uses a different cache key than other backend-based workflows because this workflows dependencies are different
+          cache-files-hash: ${{ hashFiles('requirements-pre-commit.txt') }}
+
+      - name: Setup pre-commit
+        if: needs.changes.outputs.backend == 'true'
+        env:
+          SENTRY_NO_VIRTUALENV_CREATION: 1
+        run: |
+          make setup-git
+
+      - uses: getsentry/paths-filter@v2
```

Comment:
This is the only `getsentry/paths-filter@v2` that I have not removed since I do not want to mess it up.